lib/beaker/dsl/install_utils/puppet_utils.rb

```diff
@@ -280,20 +290,28 @@ def configure_puppet_on(host, opts = {})
         # @return nil
         # @api private
         def install_puppet_from_rpm( host, opts )
-          release_package_string = "http://yum.puppetlabs.com/puppetlabs-release-#{opts[:family]}-#{opts[:release]}.noarch.rpm"
+          if version_is_less(opts[:version], '4.0.0')
+            release_package_string = "http://yum.puppetlabs.com/puppetlabs-release-#{opts[:family]}-#{opts[:release]}.noarch.rpm"
 
-          on host, "rpm -q --quiet puppetlabs-release || rpm -ivh #{release_package_string}"
+            on host, "rpm -q --quiet puppetlabs-release || rpm -ivh #{release_package_string}"
 
-          if opts[:facter_version]
-            on host, "yum install -y facter-#{opts[:facter_version]}"
-          end
+            if opts[:facter_version]
+              on host, "yum install -y facter-#{opts[:facter_version]}"
+            end
 
-          if opts[:hiera_version]
-            on host, "yum install -y hiera-#{opts[:hiera_version]}"
-          end
+            if opts[:hiera_version]
+              on host, "yum install -y hiera-#{opts[:hiera_version]}"
+            end
+
+            puppet_pkg = opts[:version] ? "puppet-#{opts[:version]}" : 'puppet'
+            on host, "yum install -y #{puppet_pkg}"
+          else
+            release_package_string = "http://yum.puppetlabs.com/puppetlabs-release-pc1-#{opts[:family]}-#{opts[:release]}.noarch.rpm"
```

Comment:
@justinstoller shouldn't we be able to pass in the repo rather than assume `pc1`?  Refer to [BKR-184](https://tickets.puppetlabs.com/browse/BKR-184) (internal only) for context.